### ISSUE TITLE
After redundantReturn is on, return is mistakenly deleted, resulting in a serious logical error.

### ISSUE THREAD

**Issue by mlch911** [nicklockwood/SwiftFormat#1739]
```swift
public func findNegativeFeedbackButton() -> NegativeFeedbackButton? {
    let btns = [topView, contentView, bottomView]
        .compactMap { $0 as? NegativeFeedbackCellProtocol }
        .map(\.negativeFeedbackButton)
    if btns.count > 1, let btn = btns.first { !$0.isHidden && $0.alpha > 0.01 } {
        return btn  // <== this return will be deleted...
    }
    return btns.first
}
```

**Comment by nicklockwood** [comment]
Sorry about that. The issue is due to the closure in the if statement condition causing the if statement body to be mistaken for a closure as well.

A workaround for now it to put parentheses around the `{ !$0.isHidden && $0.alpha > 0.01 }`

**Comment by nysander** [comment]
I had similar issue 

```
        let completion: AddItemCompletionBlock? = { [unowned self] (success: Bool, serialNumber: SerialNumber?) in
            Task { @MainActor in
                if let serialNumber,
                   let item = itemRepository.fetch(serialNumber: serialNumber) {
                    self.determineViewToShow(with: item, refresh: true)
                    self.popToHomeScreen(with: serialNumber, refresh: success)
                } else {
                    self.popToHomeScreen()
                }
            }
            return // < THIS IS REMOVED and causes crash
        }
 ```

**Comment by nicklockwood** [comment]
@nysander this issue is unrelated - can you open a separate ticket to track it please

**Comment by nicklockwood** [comment]
@nysander fixed in 0.54.1

**Comment by mlch911** [comment]
> @mlch911 fixed in 0.54.1

I installed 0.54.1, this issue is still there.😂

**Comment by mlch911** [comment]
```swift
public func tabVCIndex(path: RouterPath?, router: Router) -> (Int, UINavigationController)? {
    guard let path, let _ = path as? TabBarRouterPath, path.superPath == nil,
            let tabVC = UIApplication.shared.delegate?.window??.rootViewController as? UITabBarController
    else { return nil }
    let vcs = tabVC.viewControllers?.compactMap { $0 as? UINavigationController }
    if let path_vc = path.vc as? AnyClass, let (index, vc) = vcs?.enumerated().first { $0.element.viewControllers.first?.isMember(of: path_vc) ?? false } {
        return (index, vc)  // <== this return will be deleted...
    }
    return nil
}
```

**Comment by nicklockwood** [comment]
@mlch911 apologies, I oversimplified the test example and so the fix only covered some cases. I'll get this case fixed too

**Comment by nicklockwood** [comment]
@mlch911 fixed in 0.54.2
